crates/ruff_python_ast/src/helpers.rs

```diff
@@ -265,6 +265,24 @@ where
     }
 }
 
+
+pub fn any_over_type_param<F>(type_param: &TypeParam, func: &F) -> bool
+where
+    F: Fn(&Expr) -> bool,
+{
+    match type_param {
+        TypeParam::TypeVar(ast::TypeParamTypeVar { bound, .. }) => {
+            bound.as_ref().map_or(false, |value| any_over_expr(value, func))
+        }
+        TypeParam::TypeVarTuple(ast::TypeParamTypeVarTuple { .. }) => {
+            false
```

Comment:
Should `any_over_expr` visit names?

Reply:
It should visit all expressions, so needs to recurse on anything that might contain an expression.

Reply:
The names can't contain expressions, they're just containers for a string. It seems like a pain if you _were_ trying to visit parameter names here but I guess we can tackle that with another helper (or as it comes up).